lib/metasploit/framework/data_service/remote/http/remote_workspace_data_service.rb

```diff
@@ -20,15 +20,13 @@ def workspace
     # will not return the correct results. Run it back through the proxy.
     wlog "[DEPRECATION] Calling workspace from within the RemoteDataService is no longer supported. Please call from WorkspaceDataProxy instead."
```

Comment:
no, this logic is broken and technically should all be removed since everything is handled  via the proxy, but I don't want to affect scope anymore than it is I simply wanted to rid the unnecessary coupling to framework

Reply:
If the method remains and does nothing, shouldn't we be throwing an exception since the caller would expect changes to be made that are not?

Reply:
I agree that these probably need to just be removed. It's all in the proxy now. We should probably do it in a separate ticket because I don't think it's as simple as removing the code; I'm afraid there will be a little fallout.

Reply:
I created a ticket to address that. MS-3174